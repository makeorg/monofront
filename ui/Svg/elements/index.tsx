// @flow
export { SvgEmptyAvatar } from './EmptyAvatar';
export { SvgAngleArrowRight } from './AngleArrowRight';
export { SvgAngleArrowLeft } from './AngleArrowLeft';
export { SvgAngleArrowTop } from './AngleArrowTop';
export { SvgAngleArrowBottom } from './AngleArrowBottom';
export { SvgArrowRight } from './ArrowRight';
export { SvgArrowLeft } from './ArrowLeft';
export { SvgArrowTop } from './ArrowTop';
export { SvgPlayButton } from './PlayButton';
export { SvgThumbsUp } from './ThumbsUp';
export { SvgShare } from './Share';
export { SvgClose } from './Close';
export { SvgLightBulb } from './LightBulb';
export { SvgPencil } from './Pencil';
export { SvgFacebookLogo } from './FacebookLogo';
export { SvgFacebookLogoF } from './FacebookLogoF';
export { SvgGoogleLogoG } from './GoogleLogoG';
export { SvgLinkedinLogoIn } from './LinkedinLogoIn';
export { SvgTwitterLogo } from './TwitterLogo';
export { SvgEnvelope } from './Envelope';
export { SvgLock } from './Lock';
export { SvgExternalLink } from './ExternalLink';
export { SvgStepForward } from './StepForward';
export { SvgEye } from './Eye';
export { SvgEyeSlash } from './EyeSlash';
export { SvgCheck } from './Check';
export { SvgPaperPlane } from './PaperPlane';
export { SvgUser } from './User';
export { SvgChild } from './Child';
export { SvgMapMarker } from './MapMarker';
export { SvgMegaphone } from './Megaphone';
export { SvgSuitcase } from './Suitcase';
export { SvgClapping } from './Clapping';
export { SvgSignOut } from './SignOut';
export { SvgSaveFileOption } from './SaveFileOption';
export { SvgLike } from './Like';
export { SvgCheckedSymbol } from './CheckedSymbol';
export { SvgUsers } from './Users';
export { SvgSadEmoji } from './SadEmoji';
export { SvgInfos } from './Infos';
export { SvgDisconnect } from './Disconnect';
export { SvgAlert } from './Alert';
export { SvgSearch } from './Search';
export { SvgCalculator } from './Calculator';
export { SvgLightning } from './Lightning';
export { SvgMap } from './Map';
export { SvgInstagramLogo } from './InstagramLogo';
export { SvgTrending } from './Trending';
export { SvgLink } from './Link';
export { SvgChat } from './Chat';
export { SvgIdea } from './Idea';
export { SvgPeople } from './People';
export { SvgLight } from './Light';
export { SvgHandHeart } from './HandHeart';
export { SvgHouse } from './House';
export { SvgMail } from './Mail';
export { SvgClock } from './Clock';
export { SvgBigArrowRight } from './BigArrowRight';
export { SvgNextArrow } from './NextArrow';
export { SvgPreviousArrow } from './PreviousArrow';
export { SvgExternalLinkPlain } from './ExternalLinkPlain';
export { SvgFist } from './Fist';
export { SvgLegalConsent } from './LegalConsent';
export { SvgLogoBlack } from './LogoBlack';
export { SvgLogo } from './Logo';
export { SvgMailPlain } from './MailPlain';
export { SvgCheckedLightBulb } from './CheckedLightBulb';
export { SvgPreviousArrowLeft } from './PreviousArrowLeft';
export { SvgWorldMap } from './WorldMap';
export { SvgBigCheck } from './BigCheck';
export { SvgArrowBottom } from './ArrowBottom';
export { SvgArrowReturn } from './ArrowReturn';
export { SvgArrowUp } from './ArrowUp';
export { SvgArrowDown } from './ArrowDown';
export { SvgChevronArrowRight } from './ChevronArrowRight';
export { SvgChevronArrowLeft } from './ChevronArrowLeft';
export { SvgCookie } from './Cookie';
export { SvgLoudSpeaker } from './LoudSpeaker';
export { SvgStats } from './Stats';
export { SvgSmiley } from './Smiley';
export { SvgSettings } from './Settings';
export { SvgSwitchLabelChecked } from './SwitchLabelChecked';
export { SvgSwitchLabelCross } from './SwitchLabelCross';
export { SvgSelectArrow } from './SelectArrow';
export { SvgFastForward } from './FastForward';
