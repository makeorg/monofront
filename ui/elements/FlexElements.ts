import styled, { css } from 'styled-components';
import { intToPx } from '@make.org/utils/helpers/styled';
import { Breakpoints } from '@make.org/assets/vars/Breakpoints';

export const FlexElementStyle = styled.div`
  display: flex;
`;

export const ColumnToRowElementStyle = styled(FlexElementStyle)`
  flex-flow: column;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    flex-flow: row;
  }
`;

export const ColumnElementStyle = styled(FlexElementStyle)`
  flex-flow: column;
`;

/* Row Elements */

export const CenterRowStyle = styled(FlexElementStyle)`
  justify-content: center;
`;

export const MiddleRowStyle = styled(CenterRowStyle)`
  align-items: center;
`;

export const SpaceBetweenRowStyle = styled(FlexElementStyle)`
  justify-content: space-between;
  &.fullwidth {
    width: 100%;
  }
`;

/* Column Elements */
export const CenterColumnStyle = styled(ColumnElementStyle)`
  align-items: center;
`;

export const MiddleColumnStyle = styled(CenterColumnStyle)`
  justify-content: center;
`;

export const StartColumnStyle = styled(ColumnElementStyle)`
  justify-content: start;
`;

export const SpaceBetweenColumnStyle = styled(ColumnElementStyle)`
  justify-content: space-between;
`;

/* Column To Row Elements */
export const CenterColumnStyleToRowStyle = styled(ColumnToRowElementStyle)`
  align-items: center;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    justify-content: center;
  }
`;

export const MiddleColumnToRowStyle = styled(CenterColumnStyleToRowStyle)<{
  column?: boolean;
}>`
  justify-content: center;
  ${({ column }) =>
    column
      ? css`
          flex-direction: column;
        `
      : ''}
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    align-items: center;
  }
`;

export const SpaceBetweenColumnToRowStyle = styled(ColumnToRowElementStyle)`
  align-items: center;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    justify-content: space-between;
  }
`;
