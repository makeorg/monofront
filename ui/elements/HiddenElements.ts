import styled from 'styled-components';
import { pxToRem } from '@make.org/utils/helpers/styled';
import { Breakpoints } from '@make.org/assets/vars/Breakpoints';

export const HiddenItemStyle = styled.span`
  position: absolute;
  display: none;
  width: 0;
  height: 0;
  z-index: -100;
  visibility: hidden;
`;

export const HiddenOnTablet = styled.span`
  display: none !important;
  visibility: hidden;
  @media (min-width: ${pxToRem(Breakpoints.Desktop)}) {
    display: flex !important;
    visibility: visible;
  }
`;
