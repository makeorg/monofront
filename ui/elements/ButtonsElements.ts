import { Link } from 'react-router-dom';
import styled, { keyframes } from 'styled-components';
import { color, typography } from 'athena-design-tokens';
import { intToPx } from '@make.org/utils/helpers/styled';
import { Breakpoints, DefaultPadding } from '@make.org/assets/vars/Breakpoints';
import { MakeFonts } from '@make.org/assets/vars/Fonts';
import { SvgExternalLinkPlain } from '../Svg/elements';

export const ButtonsWrapperStyle = styled.div`
  display: flex;
  flex-flow: column;
  width: 100%;
  justify-content: center;
  align-items: center;
  margin: 10px 0;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    flex-flow: row;
    margin: 20px 0;
  }
`;

export const UnstyledButtonStyle = styled.button.attrs(props => ({
  type: 'button',
  ...props,
}))`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0;
  border: none;
  background-color: transparent;
`;

export const BasicButtonStyle = styled.button.attrs(props => ({
  type: 'button',
  ...props,
}))`
  display: flex;
  justify-content: center;
  align-items: baseline;
  font-family: ${MakeFonts.TradeGothicBoldCondensed};
  font-size: ${intToPx(typography.font.fontsize.XS.value)};
  line-height: 1;
  border: none;
  border-radius: 20px;
  text-transform: uppercase;
  padding: 12px 25px 8px;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    font-size: ${intToPx(typography.font.fontsize.XS.value)};
    padding: 13px 25px 10px;
  }
  @media (min-width: ${intToPx(Breakpoints.LargeDesktop)}) {
    font-size: ${intToPx(typography.font.fontsize.S.value)};
  }
`;

const RedStyle = `
  color: ${color.white};
  background-color: ${color.brandSecondary};
  .tofill {
    fill: ${color.white};
  }
  &:hover,
  &:focus {
    color: ${color.white};
    text-decoration: none;
  }
`;

const GreyStyle = `
  color: ${color.greyDark};
  background-color: ${color.greyLighter};
  .tofill {
    fill: ${color.brandSecondary};
  }
  &:hover,
  &:focus {
    color: ${color.greyDark};
    text-decoration: none;
  }
`;

export const RedButtonStyle = styled.button.attrs(props => ({
  type: 'button',
  ...props,
}))`
  ${BasicButtonStyle};
  ${RedStyle};
  &:disabled {
    ${GreyStyle};
  }
`;

export const GreyButtonStyle = styled.button.attrs(props => ({
  type: 'button',
  ...props,
}))`
  ${BasicButtonStyle};
  ${GreyStyle};
`;

export const LinkAsRedButtonStyle = styled(Link)`
  ${BasicButtonStyle};
  ${RedStyle};
  &:disabled {
    ${GreyStyle};
  }
`;

export const WhiteButtonStyle = styled(BasicButtonStyle)`
  color: ${color.black};
  background-color: ${color.white};
  svg {
    fill: ${color.black};
  }
`;

export const SmallRedButtonStyle = styled(RedButtonStyle)`
  padding: 5px 15px 2.5px;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    padding: 10px 20px;
  }
`;

export const SmallGreyButtonStyle = styled(GreyButtonStyle)`
  padding: 5px 15px 2.5px;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    padding: 10px 20px;
  }
`;

export const ActiveButtonStyle = styled(BasicButtonStyle)`
  color: ${color.white};
  background-color: ${color.brandSecondary};
  svg,
  .tofill {
    fill: ${color.white};
  }
  &:disabled {
    color: ${color.greyDark};
    background-color: ${color.grey};
    svg,
    .tofill {
      fill: ${color.greyDark};
    }
  }
`;

export const SocialButtonStyle = styled(ActiveButtonStyle)`
  width: 100%;
  margin: 0 5px;
  padding: 5px 15px;
  color: ${color.white};
  svg {
    fill: ${color.white};
  }
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    padding: 10px 20px;
  }
`;
export const RedLinkButtonStyle = styled(UnstyledButtonStyle)`
  display: inline-block;
  color: ${color.brandSecondary};
  text-decoration: underline;
  margin: 0 5px;
`;

export const CloseButtonStyle = styled(UnstyledButtonStyle)`
  position: absolute;
  top: ${intToPx(DefaultPadding.Mobile)};
  right: ${intToPx(DefaultPadding.Mobile)};
  fill: ${color.brandSecondary};
  z-index: 1;
  font-size: ${intToPx(typography.font.fontsize.XS.value)};
  .tofill {
    fill: ${color.greyDark};
  }
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    top: ${intToPx(DefaultPadding.Desktop)};
    right: ${intToPx(DefaultPadding.Desktop)};
  }
`;

export const NavButtonStyle = styled(UnstyledButtonStyle)`
  font-family: ${MakeFonts.TradeGothicBoldCondensed};
  font-size: ${intToPx(typography.font.fontsize.XS.value)};
  color: ${color.brandSecondary};
  text-transform: uppercase;
`;

export const QualifyButtonStyle = styled.button`
  font-family: ${MakeFonts.CircularStandardBold};
  display: flex;
  justify-content: space-between;
  width: 100%;
  border-width: 2px;
  font-size: ${intToPx(typography.font.fontsize.X2S.value)};
  line-height: 20px;
  border-style: solid;
  padding: 0 10px;
  border-radius: 36px;
  border-color: ${props => props.color};
  color: ${props => props.color};
  background-color: ${color.white};
  &.qualified {
    color: ${color.white};
    background-color: ${props => props.color};
  }
  @media (min-width: ${intToPx(Breakpoints.LargeDesktop)}) {
    font-size: ${intToPx(typography.font.fontsize.XS.value)};
    line-height: 26px;
  }
`;

const RotateButton = keyframes`
  0% { transform: rotate(0deg); }
  45% { transform: rotate(-20deg); }
  65% { transform: rotate(20deg); }
  100% { transform: rotate(0deg); }
`;

const InverseRotateButton = keyframes`
  0% { transform: rotate(0deg); }
  45% { transform: rotate(20deg); }
  65% { transform: rotate(-20deg); }
  100% { transform: rotate(0deg); }
`;

export const VoteButtonStyle = styled.button.attrs(props => ({
  type: 'button',
  ...props,
}))`
  position: relative;
  z-index: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 38px;
  height: 38px;
  min-width: 38px;
  min-height: 38px;
  border-width: 2px;
  border-style: solid;
  border-radius: 50%;
  overflow: hidden;
  padding: 0;
  background-color: ${color.white};
  transform: scale(1);
  transition: transform 0.1s ease-in;
  &:hover,
  &:focus {
    transform: scale(1.1);
  }
  &.agree {
    color: ${color.agree};
    border-color: ${color.agree};
  }
  &.disagree {
    color: ${color.disagree};
    border-color: ${color.disagree};
  }
  &.neutral,
  &.other {
    color: ${color.neutral};
    border-color: ${color.neutral};
  }
  &.animated {
    box-shadow: 0 0 0 0 ${props => props.color};
    animation: ${RotateButton} 0.5s 1;
    transform: scale(0.9);
  }
  &.animated.disagree {
    animation: ${InverseRotateButton} 0.5s 1;
  }
  &.agree.voted {
    background-color: ${color.agree};
  }
  &.disagree.voted {
    background-color: ${color.disagree};
  }
  &.neutral.voted,
  &.other.voted {
    background-color: ${color.neutral};
  }
  &.agree .tofill {
    fill: ${color.agree};
  }
  &.disagree .tofill {
    fill: ${color.disagree};
  }
  &.neutral .tofill,
  &.other .tofill {
    fill: ${color.neutral};
  }
  &.voted {
    color: ${color.white};
  }
  &.voted .tofill {
    fill: ${color.white};
  }
  &.voted:hover,
  &.voted:focus {
    transform: scale(1);
  }
  @media (min-width: ${intToPx(Breakpoints.LargeMobile)}) {
    width: 43px;
    height: 43px;
    min-width: 43px;
    min-height: 43px;
  }
  @media (min-width: ${intToPx(Breakpoints.LargeDesktop)}) {
    width: 48px;
    height: 48px;
    min-width: 48px;
    min-height: 48px;
  }
`;

export const HidePasswordIconStyle = styled(UnstyledButtonStyle)`
  color: ${color.greyDark};
`;

export const ExternalLinkIconStyle = styled(SvgExternalLinkPlain)<{
  className?: string;
}>`
  width: 14px;
  height: 14px;
  margin-left: 7px;
`;

const ButtonNoBackgroundStyle = `
  font-size: ${intToPx(typography.font.fontsize.XS.value)};
  text-decoration: underline;
  @media (min-width: ${intToPx(Breakpoints.Desktop)}) {
    font-size: ${intToPx(typography.font.fontsize.XS.value)};
  }
`;

const GreyNoBackgroundStyle = `
  color: ${color.greyDark};
  .tofill {
    fill: ${color.greyDark};
  }
`;

const RedNoBackgroundStyle = `
  color: ${color.brandSecondary};
  .tofill {
    fill: ${color.brandSecondary};
  }
`;

const BlackBordersStyle = `
  background-color: transparent;
  color: ${color.black};
  border: 1px solid ${color.black};
  .tofill {
    fill: ${color.black};
  }
`;

export const GreyNoBackgroundButtonStyle = styled(UnstyledButtonStyle)`
  ${ButtonNoBackgroundStyle};
  ${GreyNoBackgroundStyle};
`;

export const RedNoBackgroundButtonStyle = styled(UnstyledButtonStyle)`
  ${ButtonNoBackgroundStyle};
  ${RedNoBackgroundStyle};
`;

export const GreyLinkStyle = styled(Link)`
  ${ButtonNoBackgroundStyle};
  ${GreyNoBackgroundStyle};
`;

export const BlackBorderButtonStyle = styled.button.attrs(props => ({
  type: 'button',
  ...props,
}))`
  ${BasicButtonStyle};
  ${BlackBordersStyle};
`;
