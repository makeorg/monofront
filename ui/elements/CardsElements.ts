import styled from 'styled-components';
import { color, typography } from 'athena-design-tokens';
import { intToPx } from '@make.org/utils/helpers/styled';
import { ShadowColors } from '@make.org/assets/vars/Colors';
import {
  Breakpoints,
  Layouts,
  DefaultPadding,
} from '@make.org/assets/vars/Breakpoints';
import { Elements } from '@make.org/assets/vars/Elements';
import { MakeFonts } from '@make.org/assets/vars/Fonts';
import { Link } from 'react-router-dom';
import { FlexElementStyle } from './FlexElements';
import { ParagraphStyle } from './ParagraphElements';
import { SvgExternalLink, SvgLock } from '../Svg/elements';
import { LinkAsRedButtonStyle, RedButtonStyle } from './ButtonsElements';

export const CardStyle = styled.article`
  display: flex;
  flex-flow: column;
  justify-content: space-between;
  width: 100%;
  flex: 1 1 auto;
  max-width: ${intToPx(Layouts.ContainerWidth)};
  padding: ${intToPx(DefaultPadding.Mobile)};
  background-color: ${color.white};
  box-shadow: 0 1px 1px 0 ${ShadowColors.BlackZeroFiveOpacity};
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    border-radius: ${intToPx(Elements.BorderRadius)};
  }
`;

export const TallCardStyle = styled(CardStyle)`
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    max-height: 550px;
    padding: ${intToPx(DefaultPadding.Desktop)};
  }
`;

export const ParticipateCardTitleStyle = styled.h4`
  font-family: ${MakeFonts.CircularStandardBold};
  text-transform: none;
  font-size: ${intToPx(typography.font.fontsize.L.value)};
  margin: 15px 0 10px;
  @media (min-width: ${intToPx(Breakpoints.Desktop)}) {
    font-size: ${intToPx(typography.font.fontsize.XL.value)};
    margin: 15px 0;
  }
`;

export const ParticipateCardAltTitleStyle = styled.h4`
  font-family: ${MakeFonts.CircularStandardBold};
  text-transform: none;
  font-size: ${intToPx(typography.font.fontsize.S.value)};
  margin-bottom: 10px;
`;

export const ParticipateCardDescriptionStyle = styled(ParagraphStyle)`
  margin-bottom: 25px;
`;

export const ParticipateCardAltDescriptionStyle = styled(ParagraphStyle)`
  margin-bottom: 15px;
`;

export const ParticipateCardButtonStyle = styled(RedButtonStyle)`
  align-self: flex-start;
`;

export const ParticipateCardLinkAsButtonStyle = styled(LinkAsRedButtonStyle)`
  align-self: flex-start;
`;

export const ParticipateCardLinkStyle = styled(Link)`
  font-family: ${MakeFonts.TradeGothicBoldCondensed};
  font-size: ${intToPx(typography.font.fontsize.XS.value)};
  text-decoration: none;
  text-transform: uppercase;
  color: ${color.black};
  border: solid ${color.black} 1px;
  border-radius: 50px;
  width: max-content;
  height: 35px;
  padding: 15px 25px;
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const ParticipateCardExternalLinkStyle = styled.a`
  display: flex;
  align-items: center;
  font-size: ${intToPx(typography.font.fontsize.XS.value)};
  color: ${color.black};
`;

export const ParticipateCardSoonStyle = styled(FlexElementStyle)`
  align-items: center;
  font-size: ${intToPx(typography.font.fontsize.XS.value)};
  color: ${color.greyDark};
`;

export const ParticipateCardSoonIconStyle = styled(SvgLock)`
  margin-right: 10px;
`;

export const ParticipateCardExternalIconStyle = styled(SvgExternalLink)`
  width: 12px;
  height: 12px;
  margin-left: 5px;
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    width: 14px;
    height: 14px;
  }
`;

export const ParticipateCardStyle = styled.section<{
  isKeywordActive?: boolean;
}>`
  display: flex;
  flex-flow: column;
  justify-content: space-between;
  background-color: ${color.white};
  border-radius: ${intToPx(Elements.BorderRadius)};
  padding: 30px;
  &.no-padding-bottom {
    padding-bottom: 0px;
  }
  &.margin-bottom {
    margin-bottom: 20px;
  }
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    &.desktop-half {
      width: 50%;
    }

    &.desktop-padding-left {
      margin-left: ${props => (props.isKeywordActive ? '0px' : '10px')};
    }
    &.desktop-padding-right {
      margin-right: ${props => (props.isKeywordActive ? '0px' : '10px')};
    }
  }
  @media (min-width: ${intToPx(Breakpoints.Desktop)}) {
    &.margin-bottom {
      margin-bottom: 30px;
    }
    &.desktop-padding-left {
      margin-left: ${props => (props.isKeywordActive ? '0px' : '15px')};
    }
    &.desktop-padding-right {
      margin-right: ${props => (props.isKeywordActive ? '0px' : '15px')};
    }
  }
`;
