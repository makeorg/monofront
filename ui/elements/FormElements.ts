import TextareaAutosize from 'react-autosize-textarea';
import { Elements } from '@make.org/assets/vars/Elements';
import styled from 'styled-components';
import { color, typography } from 'athena-design-tokens';
import { intToPx } from '@make.org/utils/helpers/styled';
import { Breakpoints } from '@make.org/assets/vars/Breakpoints';
import { MakeFonts } from '@make.org/assets/vars/Fonts';
import { SvgExternalLinkPlain } from '../Svg/elements';
import { FlexElementStyle } from './FlexElements';
import { UnstyledButtonStyle } from './ButtonsElements';

const DataPolicyLinkStyle = `
display: inline-flex;
font-family: ${MakeFonts.CircularStandardBook};
color: ${color.brandSecondary};
text-transform: none;
text-decoration: underline;
align-items: center;
font-size: ${intToPx(typography.font.fontsize.XS.value)};
line-height: 1.31;
&:hover,
&:focus {
  color: ${color.brandSecondary};
}
`;

export const CheckboxWrapper = styled(FlexElementStyle)`
  align-items: center;
  width: 100%;
  margin-bottom: 15px;
`;

export const CheckboxLabelStyle = styled.label<{
  isBlack?: boolean;
  noFontSizeChange: boolean;
  isWidget?: boolean;
}>`
  display: inline-flex;
  color: ${props => (props.isBlack ? color.black : color.greyDark)};
  font-family: ${MakeFonts.CircularStandardBook};
  font-size: ${props =>
    props.noFontSizeChange
      ? intToPx(typography.font.fontsize.XS.value)
      : intToPx(typography.font.fontsize.X2S.value)};
  a {
    color: ${props => (props.isWidget ? color.greyDark : color.brandSecondary)};
  }
  strong {
    font-family: ${MakeFonts.CircularStandardBold};
  }
  @media (min-width: ${intToPx(Breakpoints.Tablet)}) {
    font-size: ${props =>
      props.isWidget
        ? intToPx(typography.font.fontsize.X2S.value)
        : intToPx(typography.font.fontsize.XS.value)};
  }
`;

export const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
  // Hide checkbox visually but remain accessible to screen readers.
  // Source: https://polished.js.org/docs/#hidevisually
  border: 0;
  clip: rect(0 0 0 0);
  clip-path: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`;

export const FakeCheckboxInputStyle = styled.div`
  display: inline-flex;
  position: relative;
  width: 14px;
  height: 14px;
  min-width: 14px;
  border: 1px solid ${color.grey};
  margin: 4px 8px 0 0;
  z-index: 0;
  svg {
    fill: ${color.brandSecondary};
    position: absolute;
    bottom: 5%;
    left: 25%;
    z-index: 1;
    font-size: ${intToPx(typography.font.fontsize.XS.value)};
  }
`;

export const StyledCheckbox = styled(FakeCheckboxInputStyle)<{
  checked: boolean | undefined;
}>`
  svg {
    visibility: ${props => (props.checked ? 'visible' : 'hidden')};
  }
  ${HiddenCheckbox}:focus + & {
    box-shadow: 0 0 0 2px ${color.grey};
  }
`;

export const NewWindowIconStyle = styled(SvgExternalLinkPlain)`
  width: 9px;
  height: 9px;
  padding-left: 2px;
  .tofill {
    fill: ${color.brandSecondary};
  }
  &.grey .tofill {
    fill: ${color.grey};
  }
`;

export const DataPolicyNewWindowLinkStyle = styled.a<{ isWidget?: boolean }>`
  ${DataPolicyLinkStyle};
  color: ${props => (props.isWidget ? color.greyDark : color.brandSecondary)};
  font-size: ${props =>
    props.isWidget
      ? intToPx(typography.font.fontsize.X2S.value)
      : intToPx(typography.font.fontsize.XS.value)};
  &:hover,
  &:focus {
    color: ${props => (props.isWidget ? color.greyDark : color.brandSecondary)};
  }
`;

const FormStyle = `
  display: flex;
  flex-flow: column;
  justify-content: center;
  width: 100%;
  &.hidden {
    visibility: hidden;
    display: none;
  }
`;

export const FormLeftAlignStyle = styled.form`
  ${FormStyle};
  align-items: flex-start;
`;

export const FormCenterAlignStyle = styled.form`
  ${FormStyle};
  align-items: center;
`;

export const FormRequirementsStyle = styled.p`
  font-size: ${intToPx(typography.font.fontsize.X2S.value)};
  margin: 2.5px 0 15px;
  color: ${color.greyDark};
`;

export const FakeFieldStyle = styled.div<{ hasError: boolean }>`
  display: flex;
  position: relative;
  width: 100%;
  padding: 0 15px;
  border-radius: 30px;
  background-color: ${color.white};
  border-width: 1px;
  border-style: solid;
  border-color: ${props => (props.hasError ? color.error : color.grey)};
  margin-bottom: 15px;
  &:last-child {
    margin-bottom: 0;
  }
`;

export const MiddleFakeFieldStyle = styled(FakeFieldStyle)`
  align-items: center;
`;

export const FieldWrapperStyle = styled.div`
  position: relative;
  width: 100%;
  overflow: hidden;
  > input:focus,
  > input:invalid,
  > input:required:focus,
  > input:required:not([value='']),
  > input:not([value='']),
  > textarea:not(:empty),
  > textarea:focus {
    padding-top: 15px;
    line-height: 20px;
    padding-bottom: 3px;
    & + label {
      font-size: ${intToPx(typography.font.fontsize.X2S.value)};
      line-height: 20px;
    }
  }
  > textarea:not(:empty),
  > textarea:focus {
    padding-bottom: 15px;
  }
  > input:required + label {
    font-size: ${intToPx(typography.font.fontsize.XS.value)};
    line-height: 38px;
  }
`;

export const FloatingLabelStyle = styled.label`
  font-family: ${MakeFonts.CircularStandardBook};
  color: ${color.greyDark};
  font-size: ${intToPx(typography.font.fontsize.XS.value)};
  line-height: 38px;
  font-weight: normal;
  position: absolute;
  pointer-events: none;
  left: 5px;
  top: 0;
  transition: 0.25s ease all;
  white-space: nowrap;
`;

export const FormErrorsContainerStyle = styled.div`
  width: 100%;
  padding: 15px;
  margin-bottom: 10px;
  border-radius: ${intToPx(Elements.BorderRadius)};
  background-color: ${color.brandSecondary};
  color: ${color.white};
  font-size: ${intToPx(typography.font.fontsize.X2S.value)};
`;

export const FormErrorsIntroStyle = styled.p`
  font-family: ${MakeFonts.CircularStandardBold};
  margin: 0 0 10px;
`;

export const FormErrorsListItemStyle = styled.li`
  margin: 0 0 5px;
  label {
    color: ${color.white};
    font-family: ${MakeFonts.CircularStandardBold};
    text-decoration: underline;
  }
  &:last-child {
    margin: 0;
  }
  &:before {
    content: '-  ';
  }
`;

export const CustomErrorTriggerStyle = styled(UnstyledButtonStyle)`
  display: inline-flex;
  font-family: ${MakeFonts.CircularStandardBold};
  text-decoration: underline;
  margin-left: 5px;
`;

export const InputIconStyle = styled.span`
  display: flex;
  color: ${color.brandSecondary};
  font-size: ${intToPx(typography.font.fontsize.S.value)};
  margin-right: 5px;
  svg {
    fill: ${color.brandSecondary};
  }
`;

export const CenterInputIconStyle = styled(InputIconStyle)`
  align-items: center;
`;

export const TextAreaIconStyle = styled(InputIconStyle)`
  margin-top: 10px;
`;

export const BasicInputStyle = styled.input`
  border: none;
  background: transparent;
  background-color: transparent;
  width: 100%;
  font-family: ${MakeFonts.CircularStandardBold};
  color: ${color.greyDark};
  font-size: ${intToPx(typography.font.fontsize.XS.value)};
  line-height: 38px;
  padding: 0 5px;
`;

export const FormSuccessMessageStyle = styled.p`
  display: flex;
  align-content: center;
  margin-top: 10px;
  font-size: ${intToPx(typography.font.fontsize.X2S.value)};
`;

export const FormSuccessSvgStyle = {
  display: 'inline-flex',
  fontSize: '16px',
  marginRight: '5px',
  fill: color.success,
};

export const BasicTextAreaStyle = styled(TextareaAutosize)`
  width: 100%;
  border: none;
  background: transparent;
  background-color: transparent;
  font-family: ${MakeFonts.CircularStandardBold};
  color: ${color.greyDark};
  font-size: ${intToPx(typography.font.fontsize.XS.value)};
  padding: 10px 5px;
  resize: none;
`;

export const TextAreaCounterStyle = styled.div`
  font-family: ${MakeFonts.CircularStandardBook};
  color: ${color.greyDark};
  font-size: ${intToPx(typography.font.fontsize.X2S.value)};
  line-height: 1;
  position: absolute;
  right: 5px;
  bottom: 5px;
`;
