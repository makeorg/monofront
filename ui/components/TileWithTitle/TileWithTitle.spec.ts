import React from 'react';
import renderer from 'react-test-renderer';
import snapshotDiff from 'snapshot-diff';
import { SvgInfos } from '@make.org/ui/Svg/elements';
import { TileWithTitle } from './index';

jest.mock('@make.org/ui/elements/TileWithTitle/style', () => ({
  TileWithTitleStyle: 'TileWithTitleStyle',
  TileTitleStyle: 'TileTitleStyle',
  TileSeparatorStyle: 'TileSeparatorStyle',
}));

jest.mock('@make.org/ui/Svg/elements', () => ({
  SvgInfos: 'SvgInfos',
}));

describe('TileWithTitle', () => {
  const title = 'foo';
  const children = 'bar';
  const icon = <SvgInfos />;

  it('must match the snapshot with default Props', () => {
    const component = renderer
      .create(<TileWithTitle title={title}>{children}</TileWithTitle>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must match the diff snapshot between default vs with Icon', () => {
    const defaultTile = renderer
      .create(<TileWithTitle title={title}>{children}</TileWithTitle>)
      .toJSON();
    const withIconTile = renderer
      .create(
        <TileWithTitle title={title} icon={icon}>
          {children}
        </TileWithTitle>
      )
      .toJSON();
    expect(snapshotDiff(defaultTile, withIconTile)).toMatchSnapshot();
  });
});
