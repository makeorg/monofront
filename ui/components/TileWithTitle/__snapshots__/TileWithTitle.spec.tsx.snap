// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileWithTitle must match the diff snapshot between default vs with Icon 1`] = `
Snapshot Diff:
- First value
+ Second value

  <TileWithTitleStyle>
    <TileTitleStyle
      as="h3"
    >
+     <SvgInfos />
      foo
    </TileTitleStyle>
    <TileSeparatorStyle />
    bar
  </TileWithTitleStyle>
`;

exports[`TileWithTitle must match the snapshot with default Props 1`] = `
<TileWithTitleStyle>
  <TileTitleStyle
    as="h3"
  >
    foo
  </TileTitleStyle>
  <TileSeparatorStyle />
  bar
</TileWithTitleStyle>
`;
